### PR TITLE
Add global crash search

### PR DESCRIPTION
**Pull request by johnclary**
## Associated issues

- https://github.com/cityofaustin/atd-data-tech/issues/19034

## Testing

**URL to test:** [Deploy preview](https://deploy-preview-1549--atd-vze-staging.netlify.app/#/crashes)

1. Try using the new search form with a valid crash ID (`20339647`), a temporary crash ID (`T338978`), and a crash ID that doesn't exist (`12345`)
2. Confirm that the `404` page renders when you search for a crash that doesn't exist
3. Confirm that you cannot enter any non-digit characters except `T` and `t` into the search form
4. Confirm that you can use the `enter` key to submit the search form
5. Confirm that search input is cleared after the form is submitted
6. Check out the new combined account/support menu in the top-right—it should work normally

---
#### Ship list
- [ ] Check migrations for any conflicts with latest migrations in master branch
- [ ] Confirm Hasura role permissions for necessary access
- [ ] Code reviewed
- [ ] Product manager approved

### CODE REVIEW

**Review by chiaberry — COMMENTED** [review]
I see that Frank already pointed out that once you start out with a t, you can continue with letters. 

I mistyped the first test input (that was supposed to work) and got a 404, I wish my input stuck around in the search so I could have just edited it there, instead of editing the url

I like the combined dropdown menus, the knack service request URL needs a minor update to `https://atd.knack.com/dts#new-service-request/?view_249_vars=%7B%22field_398%22%3A%22Bug%20Report%20%E2%80%94%20Something%20is%20not%20working%22%2C%22field_399%22%3A%22Vision%20Zero%20Editor%22%7D"`, to remove the `(` `)` from the product name

**Comment by johnclary** [comment]
Thank you both! I opted for Frank's improved regex that still allows extra `t`'s 🍵 

@chiaberry i'm looking for more folks' opinion on whether or not the search input should clear itself after submit

File: atd-vze/src/Components/CrashNavigationSearchForm.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { Form, Input, InputGroup, InputGroupAddon, Button } from "reactstrap";
+
+// captures any non-number except the `t` and `T` (for temp crashes)
+const CRASH_ID_EXCLUDE_CHARACTERS_REGEX = /[^(0-9tT)]*/;
```
[the remark this answers is not in the document]
Comment (johnclary) [reply atd-vze/src/Components/CrashNavigationSearchForm.js]: i mentioned this outside the thread—but i opted for `/[^\dtT]*/g` to keep the code clean. ideally we would add a validation submit/block based on the bullet-proof regex you've got there, but hopefully middle ground is bearable. thank you for your help!

File: atd-vze/src/containers/DefaultLayout/DefaultHeader.js [hunk 2]
```diff
@@ -103,26 +104,20 @@ const DefaultHeader = props => {
...
                 <i className="fa fa-lock" /> Log Out
               </DropdownItem>
-            </DropdownMenu>
-          </UncontrolledDropdown>
-          <UncontrolledDropdown nav direction="down">
-            <DropdownToggle nav>
-              <i className="fa fa-question-circle fa-2x" />
-            </DropdownToggle>
-            <DropdownMenu right>
+              {/* Support section */}
```
Comment (roseeichelmann) [inline atd-vze/src/containers/DefaultLayout/DefaultHeader.js:107]: such a nice touch!

**Review by roseeichelmann — APPROVED** [review]
love this addition so much, the regex works great now. thanks for this awesome change!

File: atd-vze/src/containers/DefaultLayout/DefaultHeader.js (hunk 2 above)
Comment (mddilley) [reply atd-vze/src/containers/DefaultLayout/DefaultHeader.js:107]: +1 - so great to have these combined!

**Review by mddilley — APPROVED** [review]
Love this quality of life upgrade! 🚀 🙌 So nice to have a spot to drop an id and go!!! 🏎️

**Review by frankhereford — APPROVED** [review]
# 🚢 💨 🙏🙏🙏 🚀

**Comment by johnclary** [comment]
thanks, y'all!

**Pull request merged**